src/main/java/avd/inf/jdm/rentmycar/domain/Ride.java

```diff
@@ -0,0 +1,60 @@
+package avd.inf.jdm.rentmycar.domain;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.*;
+import java.time.LocalDateTime;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@Entity
+@Table(name = "rides")
+public class Ride {
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+//    @Column(name = "id", nullable = false)
+    private Long id;
+
+//    @OneToOne(fetch = FetchType.LAZY)
+//    @JoinColumn(name = "user_id")
+//    private User customer;
+    @OneToOne(fetch = FetchType.EAGER)
+    @JoinColumn(name = "booking_id")
+    private Booking booking;
+    private double startRideLatitude;
+    private double startRideLongitude;
+    private double endRideLatitude;
+    private double endRideLongitude;
+
+    private LocalDateTime startDateTime;
+    private LocalDateTime endDateTime;
+
+    private double totalKilometersDriven;
+    private double maxAccelerationForce;
+
+    public void calculateBonusPointsForThisRide(Booking booking, User customer) {
+        double calculateRecklessness =  100 + ((totalKilometersDriven*1.5) + (maxAccelerationForce*10));
+        int driversAge = booking.getCustomer().calculateAge();
+
+        // drivers below age 30 get bigger penalty
+        double recklessnessToPoints = calculateRecklessness / (driversAge < 30 ? 2.5 : 2) ;
+
+        customer.setBonusPoints(customer.getBonusPoints() + (int)recklessnessToPoints);
+    };
+
+    public Ride(User customer, Booking booking, double startRideLatitude, double startRideLongitude, double endRideLatitude, double endRideLongitude, double totalKilometersDriven, double maxAccelerationForce, LocalDateTime startDateTime, LocalDateTime endDateTime) {
```

Comment:
I think not all of these properties are needed in the constructor. For example, when we create a new `Ride`, we don't know the `totalKilometersDriven` or the `maxAccelerationForce`, so we will set them later anyway.. I think we should set those using the Setters when the information is available and on the constructor only inject the properties that are already known. Or do I misunderstand constructors in Spring Boot? 🤔 😅

Reply:
Thats true, It made testing easier and also creating objects easier in the commandlinerunner. I'll change this.